mapdrawgdal.c

```diff
@@ -333,24 +333,32 @@ int msDrawRasterLayerGDAL(mapObj *map, layerObj *layer, imageObj *image,
 
   /*
    * Set up the band selection.  We look for a BANDS directive in the
-   * the PROCESSING options.  If not found we default to red=1 or
-   * red=1,green=2,blue=3 or red=1,green=2,blue=3,alpha=4.
+   * the PROCESSING options.  If not found we default to grey=1, grey=1,alpha=2,
+   * red=1,green=2,blue=3 or red=1,green=2,blue=3,alpha>=4.
    */
 
   if( CSLFetchNameValue( layer->processing, "BANDS" ) == NULL ) {
+    const int gdal_band_count = GDALGetRasterCount( hDS );
     red_band = 1;
 
-    if( GDALGetRasterCount( hDS ) >= 4
-        && GDALGetRasterColorInterpretation(
-          GDALGetRasterBand( hDS, 4 ) ) == GCI_AlphaBand )
-      alpha_band = 4;
+    if( gdal_band_count >= 4 )
+    {
+        /* The alpha band is not necessarily the 4th one */
+        for( i = 4; i <= gdal_band_count; i ++ ) {
+            if( GDALGetRasterColorInterpretation(
+                GDALGetRasterBand( hDS, i ) ) == GCI_AlphaBand ) {
```

Comment:
Are we sure that all GDAL drivers will correctly identify which band is the raster band? Should we set band==4 as the default if no band was explicitely marked as being the alpha one (with the current patch, band 0 will be used in that case)